axiom/nr_distributed_trace.c

```diff
@@ -1197,11 +1200,20 @@ char* nr_distributed_trace_create_w3c_tracestate_header(
     sampled = "1";
   }
 
-  trace_context_header = nr_formatf(
-      "%s@nr=0-0-%s-%s-%s-%s-%s-%.6f-%" PRId64, trusted_account_key, account_id,
-      app_id, NRBLANKSTR(span_id), NRBLANKSTR(txn_id), sampled,
-      nr_distributed_trace_get_priority(dt),
-      (nr_get_time() / NR_TIME_DIVISOR_MS));
+  priority = nr_distributed_trace_get_priority(dt);
+
+  priority_buf = nr_priority_double_to_str(priority);
+
+  trace_context_header
+      = nr_formatf("%s@nr=0-0-%s-%s-%s-%s-%s-%s-%" PRId64, trusted_account_key,
+                   account_id, app_id, NRBLANKSTR(span_id), NRBLANKSTR(txn_id),
+                   sampled, priority_buf, (nr_get_time() / NR_TIME_DIVISOR_MS));
+
+  nr_free(priority_buf);
+
+  nrl_verbosedebug(NRL_CAT,
+                   "Outbound W3C TraceState Context Header generated: %s",
+                   trace_context_header);
```

Comment:
`nr_formatf` can return NULL, so I'd suggest to be ready for that like this:
```suggestion
  nrl_verbosedebug(NRL_CAT,
                   "Outbound W3C TraceState Context Header generated: %s",
                   NRSAFESTR(trace_context_header));
```

Reply:
In that vein probably best to also add a `NULL` check to the create tracestate header logic as well

Reply:
[`newrelic.special=debug_cat`](https://github.com/newrelic/newrelic-php-agent/blob/72a524cd2b0717b55d77504f935ecd246ea27cc3/agent/scripts/newrelic.ini.private.template#L51) exists to limit debug logging and only log tracing related verbose debug messages when it's set. Shouldn't this debug log be also only logged when `debug_dt` special is set? I'd also be fine with not adding `debug_dt` but re-using `debug_cat`.

Reply:
I can add a check for the `debug_cat` special flag if we only want to log this when that flag is set. A `debug_dt` option would be nice but is a little out of scope for this PR and should probably be it's own effort.

Reply:
Actually, we don't have access to the Global macros that make accessing the transaction possible here- `NRPRG` and `NRTXN` live under `agent/php_newrelic.h`, and this specific function does not receive the transaction object as an argument.

Reply:
I can lift up the debug call out of this function and place it in the caller, which does have the txn object `nr_txn_create_w3c_tracestate_header`

Reply:
see changes here 6b6d6749622af92aceaf95c5f58ad3c9e12d5b18